libraries/core/src/descriptor/mod.rs

```diff
@@ -97,6 +98,11 @@ impl DescriptorExt for Descriptor {
                 }),
             };
 
+            // Calculate the final wait_for_stop value
+            let final_wait_for_stop = node.wait_for_stop.unwrap_or_else(|| node.inputs.is_empty());
```

Comment:
This only works as expected for `CoreNodeKind::Custom`. For `CoreNodeKind::Runtime`, the `inputs` list will be always empty. I think this is the reason why the c dataflow test times out on CI.

The proper approach for `CoreNodeKind::Runtime` would be to check the inputs of each operator. We also need to update https://github.com/dora-rs/dora/blob/9f7c0959981be4f7031bd4b68e8c7bf31d1e8d55/binaries/runtime/src/lib.rs#L305-L310 and make the [main runtime loop](https://github.com/dora-rs/dora/blob/9f7c0959981be4f7031bd4b68e8c7bf31d1e8d55/binaries/runtime/src/lib.rs#L149) exit once all operators exited (we don't want to keep waiting for stop if all operators exited already).